src/main/java/org/wso2/carbon/inbound/cdc/CDCPollingConsumer.java

```diff
@@ -0,0 +1,271 @@
+    private SynapseEnvironment synapseEnvironment;
+    private long scanInterval;
+    private Long lastRanTime;
+    private CDCInjectHandler injectHandler;
+    private ExecutorService executorService = null;
+    private DebeziumEngine<ChangeEvent<String, String>> engine = null;
+
+    private enum operations {create, update, delete, truncate};
+    private enum opCodes {c, u, d, t};
+
+    public CDCPollingConsumer(Properties cdcProperties, String inboundEndpointName, SynapseEnvironment synapseEnvironment,
+                              long scanInterval, String inSequence, String onErrorSeq, boolean coordination, boolean sequential) {
+        super(cdcProperties, inboundEndpointName, synapseEnvironment, scanInterval, inSequence, onErrorSeq, coordination, sequential);
+        this.cdcProperties = cdcProperties;
+        this.inboundEndpointName = inboundEndpointName;
+        this.synapseEnvironment = synapseEnvironment;
+        this.scanInterval = scanInterval;
+        this.lastRanTime = null;
+        registerHandler(new CDCInjectHandler(injectingSeq, onErrorSeq, sequential,
+                synapseEnvironment, cdcProperties));
+        setProperties();
+    }
+
+    /**
+     * Register a handler to process the file stream after reading from the
+     * source
+     *
+     * @param injectHandler
+     */
+    public void registerHandler(CDCInjectHandler injectHandler) {
+        this.injectHandler = injectHandler;
+    }
+
+    /**
+     * This will be called by the task scheduler. If a cycle execution takes
+     * more than the schedule interval, tasks will call this method ignoring the
+     * interval. Timestamp based check is done to avoid that.
+     */
+    public void execute() {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Start : CDC Inbound EP : " + inboundEndpointName);
+        }
+        // Check if the cycles are running in correct interval and start
+        // scan
+        long currentTime = (new Date()).getTime();
+        if (lastRanTime == null || ((lastRanTime + (scanInterval)) <= currentTime)) {
+            lastRanTime = currentTime;
+            poll();
+        } else if (logger.isDebugEnabled()) {
+            logger.debug(
+                    "Skip cycle since concurrent rate is higher than the scan interval : CDC Inbound EP : " + inboundEndpointName);
+        }
+        if (logger.isDebugEnabled()) {
+            logger.debug("End : CDC Inbound EP : " + inboundEndpointName);
+        }
+    }
+
+    /**
+     * Do the CDC processing operation for the given set of properties. Then inject
+     * according to the registered handler
+     */
+    public ChangeEvent<String, String> poll() {
+        logger.debug("Start : listening to DB events : ");
+        listenDataChanges();
+        logger.debug("End : Listening to DB events : ");
+        return null;
+    }
+
+    private void listenDataChanges () {
+        executorService = Executors.newSingleThreadExecutor();
```

Comment:
Since we update the executorService reference with new executor each time, I think we are loosing the previous reference. Shall we call shutdown() to avoid this leak.

Reply:
Fixed